cpu/sam0_common/periph/adc.c

```diff
@@ -77,6 +93,75 @@ static void _adc_poweroff(void)
 #endif
 }
 
+static void _setup_clock(adc_res_t res)
+{
+#ifdef CPU_SAMD21
+ /* GCLK Setup */
+    GCLK->CLKCTRL.reg = (uint32_t)(GCLK_CLKCTRL_CLKEN
+            | GCLK_CLKCTRL_GEN(ADC_GCLK_SRC) | (GCLK_CLKCTRL_ID(ADC_GCLK_ID)));
+    /* Configure CTRLB Register HERE IS THE RESOLUTION SET! */
+    ADC_DEV->CTRLB.reg = ADC_PRESCALER | res;
```

Comment:
Not part of the clock setup.

Also, `ADC_DEV->CTRLB.reg = ADC_PRESCALER;` can be set unconditionally, then you'd have

```C
#ifdef ADC_CTRLC_RESSEL
    ADC_DEV->CTRLC.reg |= res;
#else 
    ADC_DEV->CTRLB.reg |= res;
#endif
```

(Also setting `res` directly instead of using `ADC_CTRLC_RESSEL(res)`?)

Reply:
Ah `ADC_RES_10BIT` is already defined to `ADC_CTRLB_RESSEL_10BIT` etc so we don't need to use the helper macro here.